packages/nextjs/components/UserBadges.tsx

```diff
@@ -0,0 +1,36 @@
+import { useAccount } from "wagmi";
+import { IdentificationIcon, PuzzlePieceIcon } from "@heroicons/react/24/outline";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth";
+
+export const UserBadges = () => {
+  const { address } = useAccount();
+
+  const { data: isBuilderOfBatch11 } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "allowList",
+    args: [address],
+  });
+
+  const { data: contractAddress } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "yourContractAddress",
+    args: [address],
+  });
+
+  const madeCheckin = contractAddress && contractAddress !== "0x0000000000000000000000000000000000000000";
+
+  return (
+    <div className="flex gap-2 mr-2">
+      {isBuilderOfBatch11 && (
+        <div className="tooltip tooltip-bottom" data-tip="Is builder of batch 11">
+          <IdentificationIcon className="h-6 w-6 text-success" />
+        </div>
+      )}
+      {madeCheckin && (
+        <div className="tooltip tooltip-bottom" data-tip="Made checkin">
+          <PuzzlePieceIcon className="h-6 w-6 text-success" />
+        </div>
+      )}
+    </div>
```

Comment:
Maybe we could use a cascading approach, with a single symbol/text that updates dynamically depending of the status of the builder. Such as "Not batch member" and the other cases you've implemented? This might be cleaner than having multiple symbols displayed side by side. What do you think?

Reply:
Like a symbol that evolves to the next stages? I like it, I will make this change.